### PR TITLE
Bugfix model 4.0

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*: 
bugfix
- *JIRA issue*: [MIC-4331](https://jira.ihme.washington.edu/browse/MIC-4331)
- *Research reference*: 
https://github.com/ihmeuw/vivarium_research/pull/1299

### Changes and notes
 
Changed a couple things:
One, use `get_random_variable_draws` to pull pregnancy adjustment factors into the artifact in a way that comports with pre-computed RT data.

Second, the RR and PAF for Maternal Hemorrhage had a typo and was using the Maternal Disorders key instead.

### Verification and Testing

Accepted results of V&V run by RT

### CODE REVIEW

**Review by hussain-jafari — APPROVED** [review]

File: src/vivarium_gates_nutrition_optimization/components/hemoglobin.py [hunk 1]
```diff
@@ -67,13 +67,13 @@ def setup(self, builder: Builder):
         distribution_parameters = pd.concat([mean, stddev], axis=1).reset_index()
 
         self.hemorrhage_rr = builder.lookup.build_table(
-            builder.data.load(data_keys.MATERNAL_DISORDERS.RR_ATTRIBUTABLE_TO_HEMOGLOBIN),
+            builder.data.load(data_keys.MATERNAL_HEMORRHAGE.RR_ATTRIBUTABLE_TO_HEMOGLOBIN),
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization/components/hemoglobin.py:70]: Was this just a typo before?

File: src/vivarium_gates_nutrition_optimization/data/loader.py [hunk 2]
```diff
@@ -388,10 +389,10 @@ def get_hemoglobin_data(key: str, location: str) -> pd.DataFrame:
     # Add correction factors for pregnancies
     correction_params = data_values.PREGNANCY_CORRECTION_FACTORS[key]
     dist = sampling.get_norm_from_quantiles(*correction_params)
-    # random seed
-    rng = np.random.default_rng(get_hash(f"{key}_{location}"))
+    # RT has an exact set of random numbers desired, obtained via this key
+    key = "pregnancy_adjustment_factor"
     correction_factors = pd.DataFrame(
-        np.tile(dist.rvs(size=1000, random_state=rng), (len(hemoglobin_data), 1)),
+        np.tile(get_random_variable_draws(1000,(key,dist)), (len(hemoglobin_data), 1)),
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization/data/loader.py:395]: Can we talk about why this is necessary (and desirable)?

File: src/vivarium_gates_nutrition_optimization/components/hemoglobin.py (hunk 1 above)
Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/components/hemoglobin.py:70]: yes

File: src/vivarium_gates_nutrition_optimization/data/loader.py (hunk 2 above)
Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/data/loader.py]: @rmudambi I changed this to pull the values from a CSV instead, PTAL

**Review by rmudambi — APPROVED** [review]
Reminder to make sure Ali adds information about where this data comes from in the docs. Otherwise looks good to me.

**Pull request merged**